sdk/data/azcosmos/README.md

```diff
@@ -0,0 +1,202 @@
+# Microsoft Azure Cosmos DB SDK for Go, Golang
+
+## Introduction
+
+This client library enables client applications to connect to Azure Cosmos via the SQL API. Azure Cosmos is a globally distributed, multi-model database service. 
+
+Go: [cosmodb package](github.com/Azure/azure-sdk-for-go/tree/feature/cosmos/sdk/data/azcosmos)
+
+## Getting Started
+
+### Prerequisites
+
+* Go versions 1.16 or higher
+* An Azure subscription or free Azure Cosmos DB trial account
+
+Note: If you don't have an Azure subscription, create a free account before you begin.
+You can Try Azure Cosmos DB for free without an Azure subscription, free of charge and commitments, or create an Azure Cosmos DB free tier account, with the first 400 RU/s and 5 GB of storage for free. You can also use the Azure Cosmos DB Emulator with a URI of https://localhost:8081. For the key to use with the emulator, see Authenticating requests.
+
+#### Create an Azure Cosmos DB account
+
+* From the Azure portal menu or the Home page, select Create a resource.
+* On the New page, search for and select Azure Cosmos DB.
+* On the Azure Cosmos DB page, select [Create.](github.com/Azure/azure-sdk-for-go/tree/feature/cosmos/sdk/data/azcosmos)
+* In the Create Azure Cosmos DB Account page, enter the basic settings for the new Azure Cosmos account.
+
+#### Install the package
+* Install the Azure Cosmos DB SDK for Go with `go get`:
+  ```bash
+  go get -u github.com/Azure/azure-sdk-for-go/tree/feature/cosmos/sdk/data/azcosmos
+  ```
+  
+#### Authenticate the client
+
+In order to interact with the Azure CosmosDB service you'll need to create an instance of the Cosmos Client class. To make this possible you will need an URL and key of the Azure CosmosDB service.
+
+#### Create CosmosClient
+```go
+endpoint := "someEndpoint"
+key := "someKey"
+cred, _ := NewSharedKeyCredential(key)
+    client, err := NewCosmosClient(endpoint, cred, nil)
+    if err != nil {
+        t.Fatalf("Failed to create client: %v", err)
+    }
+
+    return client
+```
+
+## Examples
+
+The following section provides several code snippets covering some of the most common CosmosDB SQL API tasks, including:
+* [Create Cosmos Client](#create-cosmos-client "Create Cosmos Client")
+* [Create Database](#create-database "Create Database")
+* [Create Container](#create-container "Create Container")
+* [CRUD operation on Items](#crud-operation-on-items "CRUD operation on Items")
+
+### Create Cosmos Client
+```go
+// Create a new CosmosClient via the CosmosClientBuilder
+// It only requires endpoint and key, but other useful settings are available
+
+const (
+    CosmosDbEndpoint = "someEndpoint"
+    CosmoDbKey = "someKey"
+)
+
+cred, _ := NewSharedKeyCredential(CosmosDbEndpoint)
+    client := newCosmosClientConnection(CosmoDbKey, cred, &CosmosClientOptions{})
```

Comment:
Honor padding for the lines.
This is wrong, the client is created with NewCosmosClient. 
`newCosmosClientConnection` is an internal API